goal_src/engine/debug/menu.gc

```diff
@@ -5,3 +5,1968 @@
+                       s1-5
+                       (the-as symbol s0-2)
+                       sv-48
+                       sv-64
+                       (the-as symbol t2-1)
+                       )
+                      )
+                     )
+                    s5-0
+                    )
+                   ((= s4-0 'float-fixed-var)
+                    (set!
+                     s5-0
+                     (new
+                      'debug
+                      'debug-menu-item-var
+                      (the-as string s5-1)
+                      (the-as int (car (cdr (cdr arg1))))
+                      (the-as int (ref arg1 4))
+                      )
+                     )
+                    (let ((s4-7 debug-menu-item-var-make-float)
+                          (s3-8 (the-as debug-menu-item-var s5-0))
+                          (s2-7
+                           (debug-menu-func-decode (car (cdr (cdr (cdr arg1)))))
+                           )
+                          (s1-7
+                           (* 0.001 (the float (/ (the-as int (ref arg1 5)) 8)))
+                           )
+                          (s0-3 (ref arg1 6))
+                          )
+                     (set!
+                      sv-80
+                      (* 0.001 (the float (/ (the-as int (ref arg1 7)) 8)))
+                      )
+                     (set!
+                      sv-96
+                      (* 0.001 (the float (/ (the-as int (ref arg1 8)) 8)))
+                      )
+                     (let ((t2-2 (/ (the-as int (ref arg1 9)) 8)))
+                      (s4-7
+                       s3-8
+                       (the-as (function int) s2-7)
+                       s1-7
+                       (the-as symbol s0-3)
+                       sv-80
+                       sv-96
+                       (the-as symbol t2-2)
+                       )
+                      )
+                     )
+                    s5-0
+                    )
+                   (else
+                    (the-as debug-menu-node #f)
+                    )
+                   )
+                  )
+                 )
+      )
+     )
+    )
+   )
+  (label cfg-41)
+  s5-0
+  )
+
+;; definition for function debug-menu-find-from-template
+;; INFO: Return type mismatch object vs debug-menu.
+(defun debug-menu-find-from-template ((arg0 debug-menu-context) (arg1 pair))
+  (let ((s4-0 (the-as object (-> arg0 root-menu))))
+   (while (begin
+           (label cfg-17)
+           (and
+            s4-0
+            (type-type? (-> (the-as debug-menu-node s4-0) type) debug-menu)
+            (not (null? arg1))
+            )
+           )
+    (let ((s3-0 (-> (the-as debug-menu s4-0) items))
+          (s4-1 (car arg1))
+          )
+     (set! arg1 (cdr arg1))
+     (let ((s5-0 (car s3-0)))
+      (while (not (null? s3-0))
+       (if
+        (string= (the-as string s4-1) (-> (the-as debug-menu-item s5-0) name))
+        (begin
```

Comment:
if with a begin in it is something the decompiler should avoid... I need to fix this